datafusion/core/tests/custom_sources.rs

```diff
@@ -296,3 +300,77 @@ fn contains_empty_exec(plan: Arc<dyn ExecutionPlan>) -> bool {
         contains_empty_exec(Arc::clone(&plan.children()[0]))
     }
 }
+
+#[tokio::test]
+async fn test_get_scan_files() -> Result<()> {
```

Comment:
I think these are better tests and they show the intended use of `get_scan_files`